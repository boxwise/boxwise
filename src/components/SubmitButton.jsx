import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import CircularProgress from "@material-ui/core/CircularProgress";

const styles = theme => ({
  root: {
    position: "relative",
    display: "inline-flex"
  },
  progress: {
    position: "absolute",
    top: "50%",
    left: "50%",
    marginTop: -12,
    marginLeft: -12
  },
  submitInDialog: {},
  submitInPage: {
    marginTop: theme.spacing.unit * 2
  }
});

const SubmitButton = ({
  classes,
  children,
  isSubmitting = false,
  isInDialog = false,
  dataTestId,
  ...props
}) => (
  <div
    className={[
      classes.root,
      isInDialog ? classes.submitInDialog : classes.submitInPage
    ].join(" ")}
  >
    <Button
      variant={isInDialog ? "text" : "contained"}
      color="primary"
      type="submit"
      data-testid={dataTestId}
      disabled={isSubmitting}
      {...props}
    >
      {children}
    </Button>
    {isSubmitting && (
      <CircularProgress size={24} className={classes.progress} />
    )}
  </div>
);

export default withStyles(styles)(SubmitButton);
