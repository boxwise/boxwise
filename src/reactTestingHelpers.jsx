import React from "react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render, fireEvent } from "react-testing-library";
import { createMuiTheme, MuiThemeProvider } from "@material-ui/core/styles";
import thunk from "redux-thunk";
import { createStore, applyMiddleware } from "redux";

export { fireEvent, wait } from "react-testing-library";

// following suggestions in https://testing-library.com/docs/react-testing-library/setup#custom-render

export const typeText = (element, value) => {
  fireEvent.change(element, { target: { value } });
};

const theme = createMuiTheme({
  typography: {
    useNextVariants: true
  }
});

export function customRender(
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState = {},
    reducer = state => state
  } = {}
) {
  const store = createStore(reducer, initialState, applyMiddleware(thunk));
  return render(
    <MuiThemeProvider theme={theme}>
      <Provider store={store}>
        <Router history={history}>{ui}</Router>
      </Provider>
    </MuiThemeProvider>
  );
}

export { customRender as render };
