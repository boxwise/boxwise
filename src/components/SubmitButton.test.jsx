import React from "react";

import { render } from "reactTestingHelpers";

import SubmitButton from "./SubmitButton";

describe("<SubmitButton />", () => {
  it("renders correctly", () => {
    const { container } = render(
      <SubmitButton isSubmitting={false}>Hello</SubmitButton>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders correctly when submitting", () => {
    const { container } = render(
      <SubmitButton isSubmitting>Hello</SubmitButton>
    );
    expect(container).toMatchSnapshot("submitting");
  });

  it("renders correctly when in dialog", () => {
    const { container } = render(<SubmitButton isInDialog>Hello</SubmitButton>);
    expect(container).toMatchSnapshot("dialog");
  });
});
