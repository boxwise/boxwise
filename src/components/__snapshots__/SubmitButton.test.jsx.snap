// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitButton /> renders correctly 1`] = `
<div>
  <div
    class="SubmitButton-root-1 SubmitButton-submitInPage-4"
  >
    <button
      class="MuiButtonBase-root-31 MuiButton-root-5 MuiButton-contained-16 MuiButton-containedPrimary-17 MuiButton-raised-19 MuiButton-raisedPrimary-20"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label-6"
      >
        Hello
      </span>
      <span
        class="MuiTouchRipple-root-34"
      />
    </button>
  </div>
</div>
`;

exports[`<SubmitButton /> renders correctly when in dialog: dialog 1`] = `
<div>
  <div
    class="SubmitButton-root-84 SubmitButton-submitInDialog-86"
  >
    <button
      class="MuiButtonBase-root-114 MuiButton-root-88 MuiButton-text-90 MuiButton-textPrimary-91 MuiButton-flat-93 MuiButton-flatPrimary-94"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label-89"
      >
        Hello
      </span>
      <span
        class="MuiTouchRipple-root-117"
      />
    </button>
  </div>
</div>
`;

exports[`<SubmitButton /> renders correctly when submitting: submitting 1`] = `
<div>
  <div
    class="SubmitButton-root-41 SubmitButton-submitInPage-44"
  >
    <button
      class="MuiButtonBase-root-71 MuiButtonBase-disabled-72 MuiButton-root-45 MuiButton-contained-56 MuiButton-containedPrimary-57 MuiButton-raised-59 MuiButton-raisedPrimary-60 MuiButton-disabled-65"
      disabled=""
      tabindex="-1"
      type="submit"
    >
      <span
        class="MuiButton-label-46"
      >
        Hello
      </span>
    </button>
    <div
      class="MuiCircularProgress-root-74 MuiCircularProgress-colorPrimary-77 MuiCircularProgress-indeterminate-76 SubmitButton-progress-42"
      role="progressbar"
      style="width: 24px; height: 24px;"
    >
      <svg
        class="MuiCircularProgress-svg-79"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle-80 MuiCircularProgress-circleIndeterminate-82"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</div>
`;
